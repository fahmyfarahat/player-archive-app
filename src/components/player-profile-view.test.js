import React from "react";
import { shallow } from "enzyme";
import PlayerProfileView from "./player-profile-view";
import PlayerStatsView from "./player-stats-view";
import mockFabioProfile from "../../json/profile-333.json"
import toJson from "enzyme-to-json";

describe("PlayerProfileView - general", () => {
  let wrapper;
  beforeEach(()=> {
    const profile = {
      ...mockFabioProfile
    };
    wrapper = shallow(<PlayerProfileView profile={profile} />);
  });
  
  it("should render without crashing", () => {
    expect(wrapper).toHaveLength(1);
  });  

  it("should render 4 cols - PlayerStatsView", () => {
    expect(wrapper.find(PlayerStatsView)).toHaveLength(4);
  });

  it("should match Snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
