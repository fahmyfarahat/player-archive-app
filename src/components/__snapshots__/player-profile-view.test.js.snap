// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerProfileView - general should match Snapshot 1`] = `
<section
  className="player-sction"
>
  <Container
    fluid={false}
  >
    <ForwardRef
      noGutters={false}
    >
      <Col
        key="attack"
        lg="3"
        md="6"
      >
        <PlayerStatsView
          stats={
            Object {
              "aerialDuelsWon": "40",
              "assists": "50",
              "blocks": "6",
              "clearances": "99",
              "duelsWon": "301",
              "foulsConceded": "55",
              "foulsWon": "90",
              "gamesPlayed": "55",
              "goals": "3",
              "goalsFromInsideBox": "2",
              "goalsFromOutsideBox": "1",
              "headedGoals": "1",
              "interceptions": "50",
              "leftFootGoals": "0",
              "minutesPerGoal": "358.0",
              "minutesPlayed": "4800",
              "otherGoals": "1",
              "passingAccuracy": "70.2",
              "redCards": "9",
              "rightFootGoals": "3",
              "shotsOnTarget": "5",
              "starts": "15",
              "substitutionOff": "70",
              "substitutionOn": "6",
              "successfulCrosses": "10",
              "tacklesWon": "30",
              "totalCrosses": "15",
              "totalPasses": "8562",
              "totalShots": "11",
              "totalTackles": "11",
              "yellowCards": "41",
            }
          }
          statsToShow={
            Array [
              "minutesPerGoal",
              "goals",
              "assists",
              "shotsOnTarget",
              "totalShots",
            ]
          }
          title="attack"
        />
      </Col>
      <Col
        key="defense"
        lg="3"
        md="6"
      >
        <PlayerStatsView
          stats={
            Object {
              "aerialDuelsWon": "40",
              "assists": "50",
              "blocks": "6",
              "clearances": "99",
              "duelsWon": "301",
              "foulsConceded": "55",
              "foulsWon": "90",
              "gamesPlayed": "55",
              "goals": "3",
              "goalsFromInsideBox": "2",
              "goalsFromOutsideBox": "1",
              "headedGoals": "1",
              "interceptions": "50",
              "leftFootGoals": "0",
              "minutesPerGoal": "358.0",
              "minutesPlayed": "4800",
              "otherGoals": "1",
              "passingAccuracy": "70.2",
              "redCards": "9",
              "rightFootGoals": "3",
              "shotsOnTarget": "5",
              "starts": "15",
              "substitutionOff": "70",
              "substitutionOn": "6",
              "successfulCrosses": "10",
              "tacklesWon": "30",
              "totalCrosses": "15",
              "totalPasses": "8562",
              "totalShots": "11",
              "totalTackles": "11",
              "yellowCards": "41",
            }
          }
          statsToShow={
            Array [
              "blocks",
              "totalTackles",
              "tacklesWon",
              "interceptions",
              "clearances",
            ]
          }
          title="defense"
        />
      </Col>
      <Col
        key="passes"
        lg="3"
        md="6"
      >
        <PlayerStatsView
          stats={
            Object {
              "aerialDuelsWon": "40",
              "assists": "50",
              "blocks": "6",
              "clearances": "99",
              "duelsWon": "301",
              "foulsConceded": "55",
              "foulsWon": "90",
              "gamesPlayed": "55",
              "goals": "3",
              "goalsFromInsideBox": "2",
              "goalsFromOutsideBox": "1",
              "headedGoals": "1",
              "interceptions": "50",
              "leftFootGoals": "0",
              "minutesPerGoal": "358.0",
              "minutesPlayed": "4800",
              "otherGoals": "1",
              "passingAccuracy": "70.2",
              "redCards": "9",
              "rightFootGoals": "3",
              "shotsOnTarget": "5",
              "starts": "15",
              "substitutionOff": "70",
              "substitutionOn": "6",
              "successfulCrosses": "10",
              "tacklesWon": "30",
              "totalCrosses": "15",
              "totalPasses": "8562",
              "totalShots": "11",
              "totalTackles": "11",
              "yellowCards": "41",
            }
          }
          statsToShow={
            Array [
              "totalPasses",
              "totalCrosses",
              "passingAccuracy",
              "successfulCrosses",
            ]
          }
          title="passes"
        />
      </Col>
      <Col
        key="cards"
        lg="3"
        md="6"
      >
        <PlayerStatsView
          stats={
            Object {
              "aerialDuelsWon": "40",
              "assists": "50",
              "blocks": "6",
              "clearances": "99",
              "duelsWon": "301",
              "foulsConceded": "55",
              "foulsWon": "90",
              "gamesPlayed": "55",
              "goals": "3",
              "goalsFromInsideBox": "2",
              "goalsFromOutsideBox": "1",
              "headedGoals": "1",
              "interceptions": "50",
              "leftFootGoals": "0",
              "minutesPerGoal": "358.0",
              "minutesPlayed": "4800",
              "otherGoals": "1",
              "passingAccuracy": "70.2",
              "redCards": "9",
              "rightFootGoals": "3",
              "shotsOnTarget": "5",
              "starts": "15",
              "substitutionOff": "70",
              "substitutionOn": "6",
              "successfulCrosses": "10",
              "tacklesWon": "30",
              "totalCrosses": "15",
              "totalPasses": "8562",
              "totalShots": "11",
              "totalTackles": "11",
              "yellowCards": "41",
            }
          }
          statsToShow={
            Array [
              "successfulCrosses",
              "yellowCards",
              "foulsConceded",
            ]
          }
          title="cards"
        />
      </Col>
    </ForwardRef>
  </Container>
</section>
`;
